swaybar/ipc.c

```diff
@@ -44,6 +47,16 @@ static void ipc_parse_config(struct config *config, const char *payload) {
 		free(config->status_command);
 		config->status_command = strdup(json_object_get_string(status_command));
 	}
+	
+	if (display_mode) {
+		free(config->display_mode);
+		config->display_mode = strdup(json_object_get_string(display_mode));
+	}
```

Comment:
In my attempt at a PR for this, `hidden_state` was always used as a sole state variable for hide/unhiding the bar. if the mode isn't `hide`, it simply always holds a "visible" value. If you want to go that route (up to you of course), then a site like this where the display mode value is set is where the default value for `hidden_state` should also be set.